### ISSUE TITLE
Prefix for imported secrets

### ISSUE THREAD

**Issue by almostjulian** [MrZoidberg/VaultSharp.Extensions.Configuration#49]
I'm pulling secrets from Vault from multiple secret paths.  Unfortunately, some of those secrets uniquely named.  Could we get a new option to insert a prefix/parentPath into the secrets imported at the path?

e.g. something like:

```
{
    "ConnectionString": "secret value",
    "Option1": "secret value 2",
}
```

with code like:

```
config.AddVaultConfiguration(
        () => new VaultOptions(
            "htpp://localhost:8200",
            new KerberosAuthMethodInfo(),
            reloadOnChange: true,
            reloadCheckIntervalSeconds: 60),
        "azure/signalr",
        "secret", prefix: "SignalR" );
```

would mean the `ConnectionString` value would be accessible under `Configuration[SignalR:ConnectionString]`

**Comment by MrZoidberg** [comment]
@almostjulian please check the `BasePath` in `VaultConfigurationSource`. It should do what you want:

```
 // act
 ConfigurationBuilder builder = new ConfigurationBuilder();
 builder.AddVaultConfiguration(
     () => new VaultOptions("http://localhost:8200", "root"),
     "SignalR", // <---- this is base path, basically the prefix
     "secret", // <---- this is the mount point
     this._logger);
```

**Comment by MrZoidberg** [comment]
So, do you want to import from Vault with a prefix, or read keys with a certain prefix, I think that's what I don't get at the moment :)

**Comment by MrZoidberg** [comment]
Ok, I see. That is something possible. I don't have much time but can do some stuff over the weekend. Feel free to submit your PR if you can.

**Comment by almostjulian** [comment]
Sure thing, I'll try to submit something later this week before the weekend.

**Comment by almostjulian** [comment]
Apologies for the delay, I've created #53 for this

**Comment by MrZoidberg** [comment]
Released in 1.0.2 https://github.com/MrZoidberg/VaultSharp.Extensions.Configuration/releases/tag/1.0.2
